hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/RpcServer.java

```diff
@@ -92,7 +92,7 @@ public abstract class RpcServer implements RpcServerInterface, ConfigurationObse
   private static final String MULTI_SERVICE_CALLS = "multi.service_calls";
 
   private final boolean authorize;
-  private final boolean isOnlineLogProviderEnabled;
+  private boolean isOnlineLogProviderEnabled;
```

Comment:
set this an the other 2 to volatile. this is the typical pattern for configs updated through onConfigurationChange. we don't need to worry so much about synchronization, but the volatile is somewhat necessary because the call to onConfigurationChange is almost certain to occur in a separate thread from those that read the value.